OpenRA.Mods.Common/Activities/Hunt.cs

```diff
@@ -25,10 +25,10 @@ public class Hunt : Activity
 		public Hunt(Actor self)
 		{
 			move = self.Trait<IMove>();
-			var attack = self.Trait<AttackBase>();
+			var attacks = self.TraitsImplementing<AttackBase>();
```

Comment:
Please change this to `self.TraitsImplementing<AttackBase>().ToArray()` to avoid repeated/unnecessary lookups when the enumerable is evaluated each tick.  Please also add a comment saying that you're intentionally not checking `IsTraitDisabled` because `HasAnyValidWeapons` already does this.